app/scripts/collections/collection.directive.js

```diff
@@ -45,15 +45,15 @@ bulletApp.directive('collection', function($log){
             scope.removeBullet = function(bullet) {
                 scope.collection.removeBullet(bullet)
                 .then(function(){
-                    scope.bullets = scope.bullets.filter(b => b.id !== bullet.id);
+                    scope.collection.bullets = scope.collection.bullets.filter(b => b.id !== bullet.id);
```

Comment:
Could this happen in the logic of `removeBullet`? If so, I'm thinking you might as well.